src/main/java/Deadline.java

```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task {
+    protected String by;
```

Comment:
Strange that all your class variables are protected.
Is there a reason why they are not private?